File: 8-kyu/logical-calculator/python/solution.py
def logical_calc(array, op):
    if op == 'AND':
        return reduce(lambda x,y: x & y,array) 
    elif op == 'OR':
        return reduce(lambda x,y: x | y,array)
    else:
        return reduce(lambda x,y: x ^ y,array) 

File: 8-kyu/logical-calculator/c#/solution.cs
using System.Linq;
using System;

public class Kata
{
  public static bool LogicalCalc(bool[] array, string op)
  {
    return array.Aggregate(operatorFactory(op));
  }
  
  private static Func<bool, bool, bool> operatorFactory(string op)
  {
    Func<bool, bool, bool> opFunc;
    switch(op)
    {
      case "XOR":
        opFunc = (bool a, bool b) => a ^ b;
        break;
      case "AND":
        opFunc = (bool a, bool b) => a && b;
        break;
      case "OR":
        opFunc = (bool a, bool b) => a || b;
        break;
      default:
        goto case "OR";
    }
    return opFunc;
  }
}